modules/core/include/opencv2/core/mat.inl.hpp

```diff
@@ -584,6 +584,16 @@ Mat::Mat(const std::initializer_list<_Tp> list)
         return;
     Mat((int)list.size(), 1, traits::Type<_Tp>::value, (uchar*)list.begin()).copyTo(*this);
 }
+
+template<typename _Tp, typename> inline
+Mat::Mat(const std::initializer_list<int> sizes, const std::initializer_list<_Tp> list)
+    : flags(MAGIC_VAL | traits::Type<_Tp>::value | CV_MAT_CONT_FLAG), dims(2), rows((int)list.size()),
+      cols(1), data(0), datastart(0), dataend(0), datalimit(0), allocator(0), u(0), size(&rows), step(0)
+{
+    if(sizes.size() != 1 || *sizes.begin() == 0 || list.size() == 0)
+        return;
```

Comment:
> `sizes.size() != 1`

Why? This should be allowed to create ND matrices.

> dims(2), rows((int)list.size()), cols(1),

Should be fixed. Initialize fields for an empty `Mat`.
It is better to use C++11 [Delegating constructor](http://en.cppreference.com/w/cpp/language/initializer_list) here.

Also this check should be added:
`sizes[0]*sizes[1]*..*sizes[dims-1] == list.total()`

> return;

No. This is an error => `CV_Error` / `CV_Assert`.

Reply:
I changed it to use Mat () as Delegating constructor.

Reply:
And added check using CV_Assert.